src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs

```diff
@@ -55,6 +56,7 @@ internal class AuthenticationLevelHandler : AuthenticationHandler<Authentication
                 }
 
                 var identity = new ClaimsIdentity(claims, AuthLevelAuthenticationDefaults.AuthenticationScheme);
+                ClaimsIdentityHelper.AddIdentityToHttpRequest(Context.Request, identity, FunctionsKeyIdentityHeaderName);
```

Comment:
This would probably be better in a middleware layer, but the work was already being done here and I wasn't sure if the extra overhead of a middleware layer would be worth it.

Reply:
Agreed - it's OK here for now - we can refactor later. Perhaps add a TODO here for that